fw/cache.c

```diff
@@ -3093,10 +3125,12 @@ cache_req_process_node(TfwHttpReq *req, tfw_http_cache_cb_t action)
 			TFW_INC_STAT_BH(cache.misses);
 			goto out;
 		}
+		req->stale_ce = ce;
+		req->stale_ce_age = age;
 	}
 
-	T_DBG("Cache: service request [%p] w/ key=%lx, ce=%p",
-	      req, ce->trec.key, ce);
+	T_DBG("Cache: service request [%p] w/ key=%lx, ce=%p", req,
+	      ce->trec.key, ce);
 
 	TFW_INC_STAT_BH(cache.hits);
 
```

Comment:
I seems that there us an extra record put in case when `tfw_handle_validation_req` returns false and `stale` is true. We put record during sending 304 response (req->stale_ce is not NULL) and after put.

Reply:
I add test to your branch